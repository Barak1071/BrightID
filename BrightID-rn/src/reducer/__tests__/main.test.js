// @flow

import { mainReducer, initialState } from '../index';

import {
  USER_TRUST_SCORE,
  GROUPS_COUNT,
  SEARCH_PARAM,
  UPDATE_CONNECTIONS,
  UPDATE_USER_DATA,
  REMOVE_USER_DATA,
  USER_AVATAR,
  REFRESH_NEARBY_PEOPLE,
  PUBLICKEY2,
} from '../../actions';

describe('main reducer', () => {
  it('should return the initial state', () => {
    expect(mainReducer(undefined, {})).toMatchSnapshot();
  });

  it('should update USER_TRUST_SCORE', () => {
    expect(
      mainReducer(undefined, {
        type: USER_TRUST_SCORE,
        trustScore: '99.9',
      }),
    ).toEqual({
      ...initialState,
      trustScore: '99.9',
    });
  });

  it('should update GROUPS_COUNT', () => {
    expect(
      mainReducer(undefined, {
        type: GROUPS_COUNT,
        count: 64,
      }),
    ).toEqual({
      ...initialState,
      groupsCount: 64,
    });
  });

  it('should update SEARCH_PARAM', () => {
    expect(
      mainReducer(undefined, {
        type: SEARCH_PARAM,
        value: 'hi john',
      }),
    ).toEqual({
      ...initialState,
      searchParam: 'hi john',
    });
  });

  it('should update USER_TRUST_SCORE', () => {
    expect(
      mainReducer(undefined, {
        type: USER_TRUST_SCORE,
        trustScore: '99.9',
      }),
    ).toMatchSnapshot();
  });

  it('should update GROUPS_COUNT', () => {
    expect(
      mainReducer(undefined, {
        type: GROUPS_COUNT,
        count: 64,
      }),
    ).toMatchSnapshot();
  });

  it('should update SEARCH_PARAM', () => {
    expect(
      mainReducer(undefined, {
        type: SEARCH_PARAM,
        value: 'hi john',
      }),
    ).toMatchSnapshot();
  });
});
