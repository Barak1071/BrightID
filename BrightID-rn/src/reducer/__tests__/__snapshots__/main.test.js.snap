// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main reducer should REMOVE_USER_DATA 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "trustScore": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "trustScore": "",
  "userAvatar": "",
}
`;

exports[`main reducer should UPDATE_CONNECTIONS 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "todo...",
      "connectionDate": 1532537998586,
      "name": "Test User",
      "publicKey": Array [],
      "trustScore": "85.1",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "trustScore": "",
  "userAvatar": "",
}
`;

exports[`main reducer should UPDATE_USER_DATA 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "trustScore": "",
    },
  ],
  "groupsCount": "",
  "name": "Test User",
  "nearbyPeople": Array [],
  "publicKey": Array [],
  "publicKey2": "",
  "searchParam": "",
  "secretKey": Array [],
  "trustScore": "",
  "userAvatar": "todo...",
}
`;

exports[`main reducer should return the initial state 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "trustScore": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "trustScore": "",
  "userAvatar": "",
}
`;

exports[`main reducer should set PUBLICKEY2 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "trustScore": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": Array [],
  "searchParam": "",
  "secretKey": "",
  "trustScore": "",
  "userAvatar": "",
}
`;

exports[`main reducer should update GROUPS_COUNT 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "trustScore": "",
    },
  ],
  "groupsCount": 64,
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "trustScore": "",
  "userAvatar": "",
}
`;

exports[`main reducer should update SEARCH_PARAM 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "trustScore": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "hi john",
  "secretKey": "",
  "trustScore": "",
  "userAvatar": "",
}
`;

exports[`main reducer should update USER_TRUST_SCORE 1`] = `
Object {
  "connections": Array [
    Object {
      "avatar": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "trustScore": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "trustScore": "99.9",
  "userAvatar": "",
}
`;
